Tests: Fix incorrect distro parsing in qe_class on Fedora.

jakub-vavra-cz opened the pull request:
Fix issue introduced in https://github.com/SSSD/sssd/pull/6456

jakub-vavra-cz commented:
Let's see if the fix works.
I missed the issue because the multihost tests were failing in the PR on "configure: error: cannot run C compiled programs." so I did not have meaningful results.

justin-stephenson commented:
This fixed the issue for Fedora but still fails on centos Stream when the distro string is `CentOS Stream release 8` because there is < 6 items in the list.

Perhaps it would be better to  explicitly handle RHEL and only split the distro str there.
```
if 'Red Hat Enterprise Linux' in self.distro:
   if '6.' in self.distro.split()[6]:
            # RHEL 6 needs service command
            service_command = f"service sssd {action}"
else
   systemctl...
```

jakub-vavra-cz commented:
> This fixed the issue for Fedora but still fails on centos Stream when the distro string is `CentOS Stream release 8` because there is < 6 items in the list.
> 
> Perhaps it would be better to explicitly handle RHEL and only split the distro str there.
> 
> ```
> if 'Red Hat Enterprise Linux' in self.distro:
>    if '6.' in self.distro.split()[6]:
>             # RHEL 6 needs service command
>             service_command = f"service sssd {action}"
> else
>    systemctl...
> ```

Hi @justin-stephenson, I updated the code following Your suggestion.

justin-stephenson reviewed (approved):
Thank you for the fix.

jakub-vavra-cz commented:
Pushed PR: https://github.com/SSSD/sssd/pull/6552

* `master`
    * fc4f0399b150f5191726e2714a1a3918fee635ea - Tests: Fix incorrect distro parsing in qe_class on Fedora.
* `sssd-2-8`
    * 8e8ca9e0d9a7e5a67c9ba7e92e922277c6511f6d - Tests: Fix incorrect distro parsing in qe_class on Fedora.
